File: Java/1164.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
 
    public static void main(String[] args) throws IOException {
        Scanner leitor = new Scanner(System.in);
        int cont = leitor.nextInt();
        int num, soma;
        for (int i = 0; i < cont; i++) {
        	soma = 0;
        	num = leitor.nextInt();
        	for (int j = 1; j < num; j++) {
        		if (num % j == 0) soma += j;
        	}
        	if (soma == num) System.out.println(num + " eh perfeito");
        	else System.out.println(num + " nao eh perfeito");
        }
    }
 
}

File: C/1164.c
#include <stdio.h>

main(){
  int num,i,soma=0,x,cont;
  scanf("%d", &cont);
  
  for(i=1; i<=cont; i++){
    scanf("%d",&num);
      
      for(x=1; x<num; x++){
        if(num%x==0){
          soma+=x;}
        }
      if(soma == num){
        printf("%d eh perfeito\n", num);
        soma=0;
      }
      else{
        printf("%d nao eh perfeito\n", num);
        soma=0;
      }
    }                                                                                                                                     
  return 0;
}